new and improved custom-attribute-supporting TheWarehouse

rwcarlsen opened the pull request:
[TODO: super amazing description]

rwcarlsen committed:
Create a new singe-stop-shop warehouse and put userobjects in it

ref #10896, #11533, #6719, #6254, others?

rwcarlsen committed:
handle pre-post aux more carefully. Finalize user objects in smaller, ordered groups

This fixes the one case (in rattlesnake) where someone wrote code where
an elemental user object depends on a side user object having been
finalized first.

rwcarlsen committed:
Convert TheWarehouse to work over dynamically specifyable attributes.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/220792/) on 096ac6f wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11994/site)

This comment will be updated on new commits.

rwcarlsen committed:
add moosedocs blurb

rwcarlsen committed:
add unit tests

rwcarlsen committed:
clean up query interfaces to use const and less cloning

rwcarlsen committed:
rename some functions, minor housekeeping

rwcarlsen committed:
performance tweaks

rwcarlsen committed:
add benchmark unit test

rwcarlsen committed:
performance tweaks

rwcarlsen committed:
speed up benchmark code

also remove benchmark cout usage

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/224563/) on 099453e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 61b2e5bc5a57971cae8a2ab34ed610252c918dc4`

rwcarlsen committed:
add warehouse benchmarking test case

Requires the newly added mesh modifier to make every elem be on its own
subdomain.  This exercises the warehouse querying code paths a lot
because the compute loops run queries for every subdomain and every
element.

rwcarlsen committed:
implement attribute hashing and switch to unordered_map cache

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/225260/) on c24f18b wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 61b2e5bc5a57971cae8a2ab34ed610252c918dc4`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/225271/) on 740203f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 61b2e5bc5a57971cae8a2ab34ed610252c918dc4`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/225477/) on bc1983c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 61b2e5bc5a57971cae8a2ab34ed610252c918dc4`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/225488/) on 14eeabd wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 61b2e5bc5a57971cae8a2ab34ed610252c918dc4`

rwcarlsen committed:
fix mutex locking slopiness causing crashes, etc.

rwcarlsen committed:
more mutex cleanup

rwcarlsen committed:
hoist subdomain-queries out of per-elem calls into subdomainChanged

rwcarlsen committed:
fix bad rebase conflict resolution

rwcarlsen committed:
fix warehouse unit benchmark to not always run

rwcarlsen committed:
update doc comments and minor cleanups

rwcarlsen committed:
rename registerAttrib to registerAttributes

rwcarlsen committed:
tweak mutex locking

permcody commented:
@friedmud - Let's see if we can give this another review and possibly merge before you leave on your trip.

rwcarlsen committed:
add missing initialize call for threaded general user objects

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/243570/) on be07e9f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11994/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11994/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 3c93846985a26aae17a95b88c2dd067d7a3da5c2`

rwcarlsen committed:
fix clang format issues due to new compiler versions

rwcarlsen committed:
minor ordering tweaks

dschwen commented:
Came here to read the _super amazing description_ :-(
